### PR TITLE
🚧 Update kotlin, compose, kotest, migrate away from FreeSpec, take #3

### PR DESCRIPTION
**Pull request by pdrj**
### Summary
Updates kotlin, compose, kotest, migrates away from FreeSpec

### Technical considerations
Please have a look if the dep updates make sense, at some point I was kind of desperate when updating stuff.

### Why is it done like this and not some other way?
This worked ;) 

### Relationships

Closes #

### Pull Request checklist

- [ ] The code does not hinder the app's performance
- [ ] The code does not introduce regressions
- [ ] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Pointers
See if everything in the app still works.

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Anyone

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -35,6 +35,7 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
         adhoc {
+            manifestPlaceholders = [crashlytics: "true"]
```
Comment (daividssilverio) [inline app/build.gradle:38]: why was this added?

Comment (pdrj) [reply app/build.gradle:38]: if there's no substitution done, the build fails.

File: app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt [hunk 2]
```diff
@@ -11,73 +12,68 @@ import com.toggl.onboarding.common.domain.OnboardingAction
...
+        val mutableValue = state.toMutableValue { }
+        val spyReducer = spyk<Reducer<AppState, AppAction>>()
+        val featureAvailabilityReducer = FeatureAvailabilityReducer(spyReducer)
 
-                    featureAvailabilityReducer.reduce(mutableValue, toggleBillableAction)
+        featureAvailabilityReducer.reduce(mutableValue, toggleBillableAction)
 
-                    verify { spyReducer.reduce(mutableValue, toggleBillableAction) }
-                }
+        verify { spyReducer.reduce(mutableValue, toggleBillableAction) }
```
Comment (daividssilverio) [inline app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt:60]: this is failing locally for me 🤔

File: app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt [hunk 3]
```diff
@@ -11,73 +12,68 @@ import com.toggl.onboarding.common.domain.OnboardingAction
...
-                        workspaces = mapOf(1L to Workspace(1, "Auto created workspace", listOf(WorkspaceFeature.Pro)))
-                    )
-                    val mutableValue = state.toMutableValue { }
-                    val spyReducer = spyk<Reducer<AppState, AppAction>>()
-                    val featureAvailabilityReducer = FeatureAvailabilityReducer(spyReducer)
+    @Test
+    fun `The FeatureAvailabilityReducer, for the toggle billable action, forwards the action normally if the edited TE is in a premium workspace`() {
+        val state = AppState().copy(
+            workspaces = mapOf(1L to Workspace(1, "Auto created workspace", listOf(WorkspaceFeature.Pro)))
+        )
```
Comment (daividssilverio) [inline app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt:53]:
```suggestion
        val state = AppState().copy(
            backStack = backStackOf(Route.StartEdit(EditableTimeEntry.empty(1))),
            workspaces = mapOf(1L to Workspace(1, "Auto created workspace", listOf(WorkspaceFeature.Pro)))
        )

```

File: app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt [hunk 4]
```diff
@@ -11,73 +12,68 @@ import com.toggl.onboarding.common.domain.OnboardingAction
...
-                        workspaces = mapOf(1L to Workspace(1, "Auto created workspace", listOf()))
-                    )
-                    val mutableValue = state.toMutableValue { }
-                    val spyReducer = spyk<Reducer<AppState, AppAction>>()
-                    val featureAvailabilityReducer = FeatureAvailabilityReducer(spyReducer)
+    @Test
+    fun `The FeatureAvailabilityReducer, for the toggle billable action, does not forward the action when the edited TE is in a non-premium workspace`() {
+        val state = AppState().copy(
+            workspaces = mapOf(1L to Workspace(1, "Auto created workspace", listOf()))
+        )
```
Comment (daividssilverio) [inline app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt:67]:
```suggestion
        val state = AppState().copy(
            backStack = backStackOf(Route.StartEdit(EditableTimeEntry.empty(1))),
            workspaces = mapOf(1L to Workspace(1, "Auto created workspace", listOf()))
        )
```

File: app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt [hunk 5]
```diff
@@ -2,6 +2,7 @@ package com.toggl.domain.reducers
 
 import com.toggl.architecture.core.Reducer
 import com.toggl.architecture.core.isOrWraps
+import com.toggl.common.CoroutineTest
```
Comment (daividssilverio) [inline app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt:5]:
```suggestion
import com.toggl.common.CoroutineTest
import com.toggl.common.feature.navigation.Route
import com.toggl.common.feature.navigation.backStackOf
import com.toggl.models.domain.EditableTimeEntry
```

File: app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt (hunk 2 above)
Comment (pdrj) [reply app/src/test/java/com/toggl/domain/reducers/FeatureAvailabilityReducerTests.kt:64]: Fixed with your suggestions 👍 Why it didn't fail on bitrise is very confusing.

**Comment by pdrj** [comment]
@semanticer I've fixed what you suggested and a bit more. About the `nested` stuff - it'd take a bit longer to fix all the files, so let's maybe create an issue to do it at some point ™️.
